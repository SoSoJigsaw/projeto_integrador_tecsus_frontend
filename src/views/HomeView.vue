<template>
    <div class="container-principal">
        <h1>Dashboards</h1>
    </div>
</template>

<script lang="ts">

</script>


<style>
.container-principal {
    margin-left: -27%;
    margin-top: 7%;
    margin-right: 10%;
    margin-bottom: 5%;
}

.container-principal h1 {
    color: #776AE3;
    font-size: 45px;
    font-weight: 400;
}
</style>