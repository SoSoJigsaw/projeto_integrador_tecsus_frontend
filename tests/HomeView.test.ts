import { describe, it, expect, vi } from 'vitest'
import { mount } from '@vue/test-utils'
import HomeView from '@/views/HomeView.vue'

it('should mount the component correctly', () => {
  const wrapper = mount(HomeView)
  expect(wrapper.exists()).toBe(true)
})

it('should start with isVisible set to false', () => {
  const wrapper = mount(HomeView)
  expect(wrapper.vm.isVisible).toBe(false)
})

it('should display the main container after 1 second', async () => {
  vi.useFakeTimers()
  const wrapper = mount(HomeView)
  vi.advanceTimersByTime(1000)
  await wrapper.vm.$nextTick()
  expect(wrapper.find('.container-principal').isVisible()).toBe(true)
  vi.useRealTimers()
})

it('should initially show the house icon', () => {
  const wrapper = mount(HomeView)
  expect(wrapper.find('.fa-house').exists()).toBe(true)
})

it('should render DashAgua and DashLuz components', () => {
  const wrapper = mount(HomeView)
  expect(wrapper.findComponent(DashAgua).exists()).toBe(true)
  expect(wrapper.findComponent(DashLuz).exists()).toBe(true)
})

it('should hide the house icon after 1 second', async () => {
  vi.useFakeTimers()
  const wrapper = mount(HomeView)
  vi.advanceTimersByTime(1000)
  await wrapper.vm.$nextTick()
  expect(wrapper.find('.fa-house').exists()).toBe(false)
  vi.useRealTimers()
})

it('should display the "Dashboards" title after 1 second', async () => {
  vi.useFakeTimers()
  const wrapper = mount(HomeView)
  vi.advanceTimersByTime(1000)
  await wrapper.vm.$nextTick()
  expect(wrapper.find('.container-principal h1').text()).toContain('Dashboards')
  vi.useRealTimers()
})

it('should have a correct grid structure for dash components', () => {
  const wrapper = mount(HomeView)
  expect(wrapper.find('.container-dash').element.style.display).toBe('grid')
})

it('should react to a manual change in isVisible', async () => {
  const wrapper = mount(HomeView)
  wrapper.vm.isVisible = true
  await wrapper.vm.$nextTick()
  expect(wrapper.find('.container-principal').isVisible()).toBe(true)
})

it('should render without errors with default props', () => {
  const wrapper = mount(HomeView)
  expect(wrapper.html()).toMatchSnapshot()
})
