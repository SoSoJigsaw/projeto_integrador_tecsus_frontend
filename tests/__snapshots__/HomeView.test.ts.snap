// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomeView.vue > should render without errors with default props 1`] = `"<div class="entre-paginas"><i class="fa-solid fa-house"></i></div>"`;
